main.tf

```diff
@@ -210,6 +210,19 @@ resource "helm_release" "castai_cluster_controller" {
   }
 }
 
+resource "null_resource" "wait_for_cluster" {
+  count = var.wait_for_cluster_ready ? 1 : 0
+  depends_on = [helm_release.castai_cluster_controller, helm_release.castai_agent]
+
+  provisioner "local-exec" {
+    command = <<-EOT
+        while ! curl ${var.api_url}/v1/kubernetes/external-clusters/${castai_gke_cluster.castai_cluster.id} -H "x-api-key: ${var.api_token}" | grep '"status"\s*:\s*"ready"'; do sleep 5; done
```

Comment:
I think if we go this direction XD 5sec is too little, maybe 30s

Reply:
Also there's another issue, for example if the onboarding failed for a certain reason, this should time out after a while. 

We can just time out after 10min for example, or check which status we return when we failed to onboard.

Reply:
I changed pooling rate to 30s, added 10 min timeout.